### PR TITLE
support psych before 4.0

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: vagrant/lib/forklift/box_factory.rb [hunk 1]
```diff
@@ -42,7 +42,11 @@ def filter_boxes!
 
     def load_box_file(file)
       file = File.read(file)
-      YAML.safe_load(ERB.new(file).result, permitted_classes: [Regexp])
+      if Gem::Version.new(Psych::VERSION) < Gem::Version.new('4.0')
```
Comment (ekohl) [inline vagrant/lib/forklift/box_factory.rb:45]: In other places I checked `RUBY_VERSION >= '2.7'`. Ruby < 2.7 doesn't allow keyword arguments this way, but 2.7 does.

**Review by ekohl — APPROVED** [review]

File: vagrant/lib/forklift/box_factory.rb (hunk 1 above)
Comment (evgeni) [reply vagrant/lib/forklift/box_factory.rb:45]: But the param comes from Psych, so I prefer this ;)

Comment (ekohl) [reply vagrant/lib/forklift/box_factory.rb:45]: The param is already there in older versions. Ruby 2.7:
```console
$ scl enable tfm ruby <<EOF
> require 'yaml'
> puts Psych::VERSION
> puts YAML.safe_load('{}', permitted_classes: [Regexp])
> EOF
3.1.0
{}
```

I don't know which version added permitted_classes, but that's been there for a long time AFAIK.

Comment (evgeni) [reply vagrant/lib/forklift/box_factory.rb:45]: 3.1.0 renamed `whitelist_classes` to `permitted_classes`: https://github.com/ruby/psych/commit/682abf20f0ddbc8d83e816022953536dad80c05b
4.0.0 made it "mandatory": https://github.com/ruby/psych/commit/0767227051dbddf1f949eef512c174deabf22891

Comment (ekohl) [reply vagrant/lib/forklift/box_factory.rb:45]: Psych 4 has `ruby_version >= 2.4` so you could have an incompatible setup this way, but I'm ok with waiting until this hits us to fix it.

**Pull request merged**